TripletCriterion.cu

```diff
@@ -7,21 +7,35 @@
 __global__ void triplet_dist_kernel(const int n, const float norm,
                                     const int nb_batch, const int length,
                                     float* x, float* y) {
+
+  // Unnecessary for loop...
+  // This is EXACTLY equivalent to `if (i < n)` statement
   for (int i = blockIdx.x*blockDim.x+threadIdx.x; i < n; i += blockDim.x*gridDim.x) {
```

Comment:
I don't think it's incorrect. `i` will never get an increment, since the increment is equal to `n` and you test for `i < n`. Am I getting this wrong? I'm saying there is no edge case here at all.